packages/plugin-core/src/test/suite-integ/RandomNoteCommand.test.ts

```diff
@@ -0,0 +1,184 @@
+import { RandomNoteConfig } from "@dendronhq/common-all";
+import { CreateNoteFactory } from "@dendronhq/common-test-utils";
+import { TestEngineUtils } from "@dendronhq/engine-test-utils";
+import * as vscode from "vscode";
+import { RandomNoteCommand } from "../../commands/RandomNote";
+import { VSCodeUtils } from "../../utils";
+import { expect } from "../testUtilsv2";
+import {
+  runLegacyMultiWorkspaceTest,
+  setupBeforeAfter,
+  withConfig,
+} from "../testUtilsV3";
+
+// common template function for RandomNoteCommand testing
+function basicTest(
+  ctx: vscode.ExtensionContext,
+  noteNames: string[],
+  validateFn: () => void,
+  done: () => void,
+  includePattern?: string[],
+  excludePattern?: string[]
+) {
+  runLegacyMultiWorkspaceTest({
+    ctx,
+    preSetupHook: async ({ wsRoot, vaults }) => {
+      for (let name of noteNames) {
+        await CreateNoteFactory({ fname: name, body: "" }).create({
```

Comment:
instead of exporting note factoring, you can use [NoteTestUtilsV4](https://github.com/dendronhq/dendron/blob/dev-kevin/packages/common-test-utils/src/noteUtils.ts#L55:L55)

Reply:
cool, will adjust and remove the export.